src/Sentry.DiagnosticSource/Internals/DiagnosticSource/SentrySqlListener.cs

```diff
@@ -90,7 +90,12 @@ private void AddSpan(SentrySqlSpanType type, string operation, KeyValuePair<stri
         {
             _hub.ConfigureScope(scope =>
             {
-                if (scope.Transaction is not { } transaction)
+                if (scope.Transaction is not {} transaction)
+                {
+                    return;
+                }
+
+                if (transaction.IsSampled != true)
```

Comment:
Just wondering, why not 
```suggestion
                if (scope.Transaction is not {} transaction || transaction.IsSampled != true)
```